build/utils/yarn.go

```diff
@@ -91,12 +94,19 @@ func GetYarnExecutable() (string, error) {
 }
 
 // GetYarnDependencies returns a map of the dependencies of a Yarn project and the root package of the project.
-// The keys are the packages' values (Yarn's full identifiers of the packages), for example: '@scope/package-name@npm:1.0.0'.
+// The keys are the packages' values (Yarn's full identifiers of the packages), for example: '@scope/package-name@1.0.0'
+// (for yarn v < 2.0.0) or @scope/package-name@npm:1.0.0 (for yarn v >= 2.0.0).
 // Pay attention that a package's value won't necessarily contain its version. Use the version in package's details instead.
-// Note that in some versions of Yarn, the version of the root package is '0.0.0-use.local', instead of the version in the package.json file.
 func GetYarnDependencies(executablePath, srcPath string, packageInfo *PackageInfo, log utils.Log) (dependenciesMap map[string]*YarnDependency, root *YarnDependency, err error) {
-	// Run 'yarn info'
-	responseStr, errStr, err := runYarnInfo(executablePath, srcPath)
+	executableVersion, err := getVersion(executablePath, srcPath)
+	if err != nil {
+		return
+	}
+
+	isV2AndAbove := strings.Compare(executableVersion, YarnV2Version) >= 0
```

Comment:
you can't compare versions with strings.compare.
here is an example why not - `isV2AndAbove := strings.Compare("11.0.0", "2.0.0") >= 0`
you should check our gofrog func (v *Version) Compare(ver1 string) function
or just to take the first number on the version string `"x.y.z" > x` , and just see if its more then 2.

Reply:
having said that, why not just use - validateYarnVersion same as today but without returning an error.

Reply:
I tried to use validateYarnVersion
validateYarnVersion is located in build-info-go/build/yarn.go and the file I need it for is build-info-go/build/utils/yarn.go which is being imported by build-info-go/build/yarn.go. this creates a circular import and will not compile